Frameworks/CoreText/CTFont.mm

```diff
@@ -692,13 +704,50 @@ CGGlyph CTFontGetGlyphWithName(CTFontRef font, CFStringRef glyphName) {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Caveat
+ @Notes  kCTFontVerticalOrientation is unsupported, returns the same as default and horizontal orientations
 */
 CGRect CTFontGetBoundingRectsForGlyphs(
     CTFontRef font, CTFontOrientation orientation, const CGGlyph glyphs[], CGRect* boundingRects, CFIndex count) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (!glyphs || count <= 0) {
+        TraceError(g_logTag, L"No glyphs were specified");
+        return CGRectNull;
+    }
+
+    if (boundingRects) {
+        if (FAILED(_DWriteFontGetBoundingBoxesForGlyphs(
+                font->_dwriteFontFace, glyphs, boundingRects, count, (orientation == kCTFontVerticalOrientation)))) {
+            TraceError(g_logTag, L"Unable to get glyph bounding boxes");
+            return CGRectNull;
+
+        } else {
+            CGRect ret = { { std::numeric_limits<CGFloat>::max(), std::numeric_limits<CGFloat>::max() },
+                           { std::numeric_limits<CGFloat>::lowest(), std::numeric_limits<CGFloat>::lowest() } };
+            for (size_t i = 0; i < count; ++i) {
+                // DWrite uses design units, where as CTFont uses point size
+                // Scale all the boundingRects
+                boundingRects[i].origin.x = __CTFontScaleMetric(font, boundingRects[i].origin.x);
+                boundingRects[i].origin.y = __CTFontScaleMetric(font, boundingRects[i].origin.y);
+                boundingRects[i].size.width = __CTFontScaleMetric(font, boundingRects[i].size.width);
+                boundingRects[i].size.height = __CTFontScaleMetric(font, boundingRects[i].size.height);
+
+                // Find the overall bounding rect
+                // This can probably be made more efficient by putting all of each attribute in an array and sorting it,
+                // but that seems like overkill for now
+                ret.origin.x = std::min(ret.origin.x, boundingRects[i].origin.x);
+                ret.origin.y = std::min(ret.origin.y, boundingRects[i].origin.y);
+                ret.size.width = std::max(ret.size.width, boundingRects[i].size.width);
+                ret.size.height = std::max(ret.size.height, boundingRects[i].size.height);
+            }
+            return ret;
+        }
+
+    } else {
```

Comment:
it reads cleaner to me with this else, I'd like to keep it

---
In reply to: [83339432](https://github.com/Microsoft/WinObjC/pull/1145#discussion_r83339432) [](ancestors = 83339432)